generated/nidaqmx/_library_interpreter.py

```diff
@@ -0,0 +1,2148 @@
+# Do not edit this file; it was automatically generated.
+
+import ctypes
+import numpy
+
+from nidaqmx import BaseInterpreter
+from nidaqmx._lib import lib_importer, ctypes_byte_str, c_bool32, wrapped_ndpointer
+from nidaqmx.errors import check_for_error
+
+class LibraryInterpreter(BaseInterpreter):
+    """
+    Library C<->Python interpreter.
+    This class is responsible for interpreting the Library's C API.
+
+    """
+
+    def calculate_reverse_poly_coeff(
+            self, forward_coeffs, num_forward_coeffs_in, reverse_coeffs,
+            min_val_x, max_val_x, num_points_to_compute, reverse_poly_order):
+
+        cfunc = lib_importer.windll.DAQmxCalculateReversePolyCoeff
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        ctypes_byte_str,
+                        wrapped_ndpointer(dtype=numpy.float64,
+                        flags=('C','W')), ctypes.c_uint, ctypes.c_double,
+                        ctypes.c_double, ctypes.c_int, ctypes.c_int,
+                        wrapped_ndpointer(dtype=numpy.float64,
+                        flags=('C','W'))]
+
+        error_code = cfunc(
+            forward_coeffs, len(forward_coeffs), num_forward_coeffs_in,
+            min_val_x, max_val_x, num_points_to_compute, reverse_poly_order,
+            ctypes.byref(reverse_coeffs))
+        check_for_error(error_code)
+        return reverse_coeffs
+
+    def cfg_anlg_edge_ref_trig(
+            self, task, trigger_source, pretrigger_samples, trigger_slope,
+            trigger_level):
+
+        cfunc = lib_importer.windll.DAQmxCfgAnlgEdgeRefTrig
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        lib_importer.task_handle, ctypes_byte_str,
+                        ctypes.c_int, ctypes.c_double, ctypes.c_uint]
+
+        error_code = cfunc(
+            task, trigger_source, trigger_slope, trigger_level,
+            pretrigger_samples)
+        check_for_error(error_code)
```

Comment:
FYI, one notable difference from nimi-python is that DAQmx uses `ctypes_byte_str` to encode all string inputs as ASCII, but nimi-python uses `create_string_buffer` to convert to a specific encoding: https://github.com/ni/nimi-python/blob/master/generated/nidcpower/nidcpower/_library_interpreter.py#L98

Fixing this is outside the scope of this project. If one did fix it, the best fix would be different from nimi-python: load the UTF-8 version of the DAQmx C API (`nicai_utf8.dll`) and encode all string inputs as UTF-8.